security/pkg/pki/ca/controller/secret.go

```diff
@@ -286,8 +286,8 @@ func (sc *SecretController) scrtDeleted(obj interface{}) {
 	}
 
 	saName := scrt.Annotations[ServiceAccountNameAnnotationKey]
-	if sa, _ := sc.core.ServiceAccounts(scrt.GetNamespace()).Get(saName, metav1.GetOptions{}); sa != nil {
```

Comment:
Thanks @myidpt for the fix. One question, have you ever test the case of keeping the sa, but just delete the secret, does the secret will be re-created with the fix?

Reply:
Yes. https://github.com/istio/istio/blob/release-1.0/security/pkg/pki/ca/controller/secret_test.go#L253